pygeos/test/test_io.py

```diff
@@ -349,3 +350,11 @@ def test_from_shapely_error(geom):
 def test_from_shapely_incompatible_versions():
     with pytest.raises(ImportError):
         pygeos.from_shapely(point)
+
+
+@pytest.mark.parametrize("geom", all_types)
+def test_pickle(geom):
+    if pygeos.get_type_id(geom) == 2:
```

Comment:
Suggestion: instead of expecting to fail on this case, how about expecting the expected output (in this case linear ring => linestring):
```
    unpickled = pickle.loads(pickle.dumps(geom))

    if pygeos.get_type_id(geom) == 2:
        # linear rings are converted to linestrings on unpickling
        expected = pygeos.linestrings(pygeos.get_coordinates(geom))
        assert pygeos.equals_exact(unpickled, expected)

    else:
        assert pygeos.equals_exact(unpickled, geom)
```

That way we know it works correctly in all cases, but that a defined case produces different output.